storages/backends/s3boto3.py

```diff
@@ -0,0 +1,560 @@
+        # But for the root path "" we leave it as an empty string
+        if name and not name.endswith('/'):
+            name += '/'
+
+        files = []
+        dirs = set()
+        base_parts = name.split("/")[:-1]
+        for item in self.bucket.objects.filter(Prefix=self._encode_name(name)):
+            parts = item.key.split("/")
+            parts = parts[len(base_parts):]
+            if len(parts) == 1:
+                # File
+                files.append(parts[0])
+            elif len(parts) > 1:
+                # Directory
+                dirs.add(parts[0])
+        return list(dirs), files
+
+    def size(self, name):
+        name = self._normalize_name(self._clean_name(name))
+        if self.entries:
+            entry = self.entries.get(name)
+            if entry:
+                return entry.content_length
+            return 0
+        return self.bucket.Object(self._encode_name(name)).content_length
+
+    def modified_time(self, name):
+        name = self._normalize_name(self._clean_name(name))
+        entry = self.entries.get(name)
+        # only call self.bucket.Object() if the key is not found
+        # in the preloaded metadata.
+        if entry is None:
+            entry = self.bucket.Object(self._encode_name(name))
+        return entry.last_modified
+
+    def _strip_signing_parameters(self, url):
+        # Boto3 does not currently support generating URLs that are unsigned. Instead we
+        # take the signed URLs and strip any querystring params related to signing and expiration.
+        # Note that this may end up with URLs that are still invalid, especially if params are
+        # passed in that only work with signed URLs, e.g. response header params.
+        # The code attempts to strip all query parameters that match names of known parameters
+        # from v2 and v4 signatures, regardless of the actual signature version used.
+        split_url = urlparse.urlsplit(url)
+        qs = urlparse.parse_qsl(split_url.query, keep_blank_values=True)
+        blacklist = set(['x-amz-algorithm', 'x-amz-credential', 'x-amz-date',
+                         'x-amz-expires', 'x-amz-signedheaders', 'x-amz-signature',
+                         'x-amz-security-token', 'awsaccesskeyid', 'expires', 'signature'])
+        filtered_qs = ((key, val) for key, val in qs if key.lower() not in blacklist)
+        # Note: Parameters that did not have a value in the original query string will have
+        # an '=' sign appended to it, e.g ?foo&bar becomes ?foo=&bar=
+        joined_qs = ('='.join(keyval) for keyval in filtered_qs)
+        split_url = split_url._replace(query="&".join(joined_qs))
+        return split_url.geturl()
+
+    def url(self, name, parameters=None, expire=None):
+        # Preserve the trailing slash after normalizing the path.
+        # TODO: Handle force_http=not self.secure_urls like in s3boto
+        name = self._normalize_name(self._clean_name(name))
+        if self.custom_domain:
```

Comment:
@mbarrien This skips `generate_presigned_url` for custom domains, which means that for those we lose the `AWS_QUERYSTRING_AUTH` parameter. I don't think that's intended.